### PR TITLE
[LMLayer] Rewrite worker code: Stage 1: constructor and onMessage()

### PR DESCRIPTION
**Pull request by eddieantonio**
**NOTE**: Review #1324 before reviewing this PR!

I deleted `lmlayer.ts`, because it was too difficult to unit test. Instead, I'm starting from scratch in a file called `./worker/index.ts`, which can be unit tested without being instantiated inside a Worker. Here's the initial work on that.

I'll be merging in functionality from the old `lmlayer.ts` in future versions PRs.

### CODE REVIEW

File: common/predictive-text/worker/tsconfig.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
   "compilerOptions": {
     "allowJs": false,
-    "module": "none",
-    "outFile": "./index.js",
+    "module": "commonjs",
```
Comment (jahorton) [inline common/predictive-text/worker/tsconfig.json]: These options need to be left unchanged - everything for "production" must be able to compile without use of modules and/or `require` statements.

I won't complain about use of those within testing environments, though.

Comment (eddieantonio) [reply common/predictive-text/worker/tsconfig.json]: Done!

**Comment by eddieantonio** [comment]
I undid the changes to tsconfig.json and rebase'd this branch on the changes made to #1324

File: common/predictive-text/unit_tests/headless/worker-intialization.js [hunk 2]
```diff
@@ -1,11 +1,59 @@
...
+  function defineLMLayerWorkerWithinSandbox(self) {
+    var fs = require('fs');
+    var sourceCode = fs.readFileSync(require.resolve('../../worker'));
+    var sandbox = `(function (self) {
+      var exports = {}; // TypeScript CommonJS code generation will attempt to write to this.
+      var postMessage = self.postMessage;
+      ${sourceCode}
+      return LMLayerWorker;
+    })`;
+    return eval(sandbox)(self);
```
[the remark this answers is not in the document]
Comment (jahorton) [reply common/predictive-text/unit_tests/headless/worker-intialization.js]: Also, minor gripe, but maybe fix the file name?

worker-intialization.js -> worker-in**i**tialization.js

[the remark this answers is not in the document]
Comment (eddieantonio) [reply common/predictive-text/unit_tests/headless/worker-intialization.js]: > While this is probably fine for testing, I am a touch iffy about it, especially if the plan's to use this pattern a lot.

I only plan on doing that for testing, to have a mocked "Worker" global, that we can control. That said, I think in a future PR I'll ditch the eval or the `Function(code)` pattern, and put tests like these in a browser/integration test.

**Pull request merged**